src/lib/features/project/project-owners-read-model.ts

```diff
@@ -122,30 +129,31 @@ export class ProjectOwnersReadModel {
         const usersDict = await this.getAllProjectUsersByRole(ownerRole.id);
         const groupsDict = await this.getAllProjectGroupsByRole(ownerRole.id);
 
-        const projects = [
-            ...new Set([...Object.keys(usersDict), ...Object.keys(groupsDict)]),
-        ];
-
-        const dict = Object.fromEntries(
-            projects.map((project) => {
-                return [
-                    project,
-                    [
-                        ...(usersDict[project] || []),
-                        ...(groupsDict[project] || []),
-                    ],
-                ];
-            }),
-        );
+        const dict: Record<
+            string,
+            Array<UserProjectOwner | GroupProjectOwner>
+        > = {};
+
+        Object.keys(usersDict).forEach((project) => {
+            dict[project] = usersDict[project];
+        });
+
+        Object.keys(groupsDict).forEach((project) => {
+            if (project in dict) {
+                dict[project] = [...dict[project], ...groupsDict[project]];
+            } else {
+                dict[project] = groupsDict[project];
+            }
+        });
```

Comment:
Nice! Still a few small things that I think can make this faster:
1. Why do we create a new dict? If we reuse the users dictionary, then we skip an iteration of those keys and save on some memory space. Yes, mutation can be iffy, but this feels like a controlled place.
2. I think `concat` is faster than spreading and two arrays into a new one (at least from my limited reading), so I'd suggest going with that instead.

Thoughts?

Reply:
Ok, I'll update it. I find it more readable in a previous form. I don't think the optimization is worth it